### PR TITLE
fix: use its own lock in serverConfigV17

### PR DESCRIPTION
**Pull request by balamurugana**
Previously serverConfigV17 used a global lock that made any instance of
serverConfigV17 depended on single global serverConfigMu.

This patch fixes by having individual lock per instances.

## How Has This Been Tested?

`make test`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/gateway-main.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func newGatewayLayer(backendType, accessKey, secretKey string) (GatewayLayer, er
 //
 // DO NOT save this config, this is meant to be
 // only used in memory.
-func newGatewayConfig(accessKey, secretKey, region string) error {
```
Comment (harshavardhana) [inline cmd/gateway-main.go]: why is this changed? this is needed when we bring the S3 backend it was kept as a placeholder so that region is addressed properly for S3 backends. 

Please keep it the way it was and only move `createConfigDir()` outside.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/config-v17.go [hunk 2]
```diff
@@ -26,17 +26,21 @@ import (
...
-// Read Write mutex for safe access to ServerConfig.
-var serverConfigMu sync.RWMutex
-
 // Config version
 const v17 = "17"
 
+var (
+	// serverConfig server config.
+	serverConfig      *serverConfigV17
+	serverConfigMutex sync.RWMutex
```
Comment (krishnasrinivas) [inline cmd/config-v17.go]: in Go, a general convention that I have seen people follow is - for object named xxxx - if it needs lock then the lock is named as xxxxMu

File: cmd/config-v17.go [hunk 3]
```diff
@@ -91,67 +162,25 @@ func newConfig() error {
...
 	}
-
 	if globalIsEnvBrowser {
 		srvCfg.SetBrowser(globalIsBrowserEnabled)
 	}
 
-	// Create config path.
-	if err := createConfigDir(); err != nil {
-		return err
-	}
```
Comment (krishnasrinivas) [inline cmd/config-v17.go:106]: I see that `createConfigDir()` is getting called in gateway mode, but unable to see it getting called for server mode

Comment (harshavardhana) [reply cmd/config-v17.go:106]: Its called in server-main.go

File: cmd/gateway-main.go (hunk 1 above)
Comment (balamurugana) [reply cmd/gateway-main.go]: Done.

Why not add when needed than unused parameter? I think you want default to be passed and set in the config struct even though new config carries the same.

File: cmd/config-v17.go (hunk 2 above)
Comment (balamurugana) [reply cmd/config-v17.go]: Done

**Review by harshavardhana — APPROVED** [review]
